### PR TITLE
Upgrade Scala to 2.12.4 and upgrade dependencies

### PR DESCRIPTION
**Pull request by leahxu**
@ravwojdyla

### CODE REVIEW

**Comment by ravwojdyla** [comment]
We would need to configure a cross build for both scala 2.11.x and 2.12.x, you could take a look at featran or scio to see an example of cross build setup.

**Comment by leahxu** [comment]
Thanks @ravwojdyla , 2.12 to the crossbuild

File: .travis.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 language: scala
 scala:
-  - 2.11.11
```
Comment (ravwojdyla) [inline .travis.yml]: Could you specify both `2.11.11` and `2.12.4`?

File: .travis.yml [hunk 2]
```diff
@@ -31,5 +31,5 @@ deploy:
   file: "ratatool/target/scala-*/ratatool-*.jar"
   skip_cleanup: true
   on:
-    scala: 2.11.11
+    scala: 2.12.4
```
Comment (ravwojdyla) [inline .travis.yml]: Let's keep 2.11.11 for the ratatool artifact tool for now.

**Comment by ravwojdyla** [comment]
Thanks @leahxu !

**Pull request merged**